QRCoder/QRCodeGenerator.cs

```diff
@@ -1041,61 +1129,64 @@ private static string PlainTextToBinary(string plainText, EncodingMode encMode,
                 case EncodingMode.Byte:
                     return PlainTextToBinaryByte(plainText, eciMode, utf8BOM, forceUtf8);
                 case EncodingMode.Kanji:
-                    return string.Empty;
                 case EncodingMode.ECI:
                 default:
-                    return string.Empty;
+                    return _emptyBitArray;
             }
         }
 
-        private static string PlainTextToBinaryNumeric(string plainText)
+        private static readonly BitArray _emptyBitArray = new BitArray(0);
+
+        private static BitArray PlainTextToBinaryNumeric(string plainText)
         {
-            var codeText = string.Empty;
-            while (plainText.Length >= 3)
+            var bitArray = new BitArray(plainText.Length / 3 * 10 + (plainText.Length % 3 == 1 ? 4 : plainText.Length % 3 == 2 ? 7 : 0));
+            var index = 0;
+            for (int i = 0; (i + 2) < plainText.Length; i += 3)
```

Comment:
```suggestion
            for (int i = 0; i < plainText.Length - 2; i += 3)
```
`i + 2` needs to be evaluated at each iteration, so `plainText.Length - 2` can be kept in a register.